data_model/src/query/parameters.rs

```diff
@@ -56,22 +55,33 @@ mod model {
         Deserialize,
         Serialize,
         IntoSchema,
+        Constructor,
     )]
     #[display(
         fmt = "{}--{}",
-        "offset.map(NonZeroU64::get).unwrap_or(0)",
+        "offset",
         "limit.map_or(\".inf\".to_owned(), |n| n.to_string())"
     )]
     pub struct Pagination {
-        /// limit of indexing
-        pub limit: Option<NonZeroU32>,
         /// start of indexing
-        pub offset: Option<NonZeroU64>,
+        pub offset: u64,
+        /// limit of indexing
+        pub limit: Option<NonZeroU64>,
```

Comment:
why reorder?

Reply:
The idea was that the `offset` is primary and `limit` is on top of it.

But

1. Breaks schema a bit
2. SQL uses LIMIT and then OFFSET

So I will revert.

Reply:
yes, the feature is known as `LIMIT OFFSET`

Reply:
Reverted